### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by mguidon**
Part 1 of comp.backend reintegration:
- Shared libraries and some test examples.

### CODE REVIEW

File: modules/s3wrapper/src/s3wrapper/s3_client.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        self.remove_objects(bucket_name, to_del)
+
+
+    def create_bucket(self, bucket_name, delete_contents_if_exists=False):
+        try:
+            if not self.exists_bucket(bucket_name):
+                self.client.make_bucket(bucket_name)
+            elif delete_contents_if_exists:
+                if self.__remove_objects_recursively(bucket_name):
+                    self.remove_bucket(bucket_name)
```
Comment (sanderegg) [inline modules/s3wrapper/src/s3wrapper/s3_client.py]: should it not re/create the bucket after removing it? 
remove_bucket already calls __remove_objects_recursively but does not check the return value

File: modules/s3wrapper/src/s3wrapper/s3_client.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    def __init__(self, endpoint, access_key=None, secret_key=None, secure=False):
+        self.__metadata_prefix = "x-amz-meta-"
+        self.client = None
+        try:
+            self.client = Minio(endpoint,
+                access_key=access_key,
+                secret_key=secret_key,
+                secure=secure)
+        except ResponseError as err:
+            print(err)
```
Comment (sanderegg) [inline modules/s3wrapper/src/s3wrapper/s3_client.py]: we started using the logging module for printing stuff [logging](https://github.com/ITISFoundation/osparc-simcore/blob/8e928ee91058e0954d28513077532bc89de4f36e/services/director/source/director.py#L13)

File: .travis.yml [hunk 3]
```diff
@@ -15,11 +15,65 @@ matrix:
...
+        - pip3 install -r modules/simcore_sdk/models/requirements.txt
+        - pip3 --version
+      
+      # Todo MaG: I have to move everything into one docker-compose.yml due to conftest.py issues
       before_script:
         - pylint --version
+        # change python path for 2nd party libraries
+        - export PYTHONPATH=${PYTHONPATH}:${TRAVIS_BUILD_DIR}/modules/s3wrapper/src
+        # pre-cache docker images to avoid time-outs later on while the tests turn
+        - docker-compose -f modules/pytest_docker/tests/docker-compose.yml pull
```
Comment (pcrespov) [inline .travis.yml]: Why pull and then build and not just build the image??

File: modules/s3wrapper/src/s3wrapper/s3_client.py (hunk 2 above)
Comment (pcrespov) [reply modules/s3wrapper/src/s3wrapper/s3_client.py]: try here ``_LOGGER.exception( ... ) ``

File: .travis.yml (hunk 3 above)
Comment (mguidon) [reply .travis.yml]: Hi Pedro
Thanks for the suggestions. Whats wrong with the linting? Am I not using the correct configuration? I get a straight 10.0...

**Pull request merged**